cmd/ztest/ztest.c

```diff
@@ -2852,7 +2858,8 @@ ztest_vdev_aux_add_remove(ztest_ds_t *zd, uint64_t id)
 
 	path = umem_alloc(MAXPATHLEN, UMEM_NOFAIL);
 
-	if (ztest_random(2) == 0) {
+	/* The kernel does not accept files for l2arc cache devices */
+	if ((ztest_random(2) == 0) || (ztest_opts.zo_kernel_pools)) {
```

Comment:
@behlendorf I'm not sure of the history behind that restriction.  Why do we now need the `-K` option?  It sounds like ztest was allowed to use file-based cache devices, was that changed?